### PR TITLE
Data v2

### PR DESCRIPTION
**Pull request by noramehesz**

### CODE REVIEW

**Review by gnvk — APPROVED** [review]
LGTM

File: lib/resources/datav2/websockets_v2.js [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    this.emit(STATE.AUTHENTICATING);
+    this.currentState = STATE.AUTHENTICATING;
+  }
+
+  onDisconnect(fn) {
+    this.on(EVENT.DISCONNECTED, () => fn());
+  }
+
+  disconnect() {
+    this.emit(EVENT.DISCONNECTED);
```
Comment (emileindik) [inline lib/resources/datav2/websockets_v2.js:317]: This should use `STATE.DISCONNECTED`

File: lib/resources/datav2/websockets_v2.js [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+      key: this.session.apiKey,
+      secret: this.session.secretKey,
+    };
+    this.conn.send(JSON.stringify(authMsg));
+    this.emit(STATE.AUTHENTICATING);
+    this.currentState = STATE.AUTHENTICATING;
+  }
+
+  onDisconnect(fn) {
+    this.on(EVENT.DISCONNECTED, () => fn());
```
Comment (emileindik) [inline lib/resources/datav2/websockets_v2.js:313]: This should use `STATE.DISCONNECTED`

File: lib/resources/datav2/websockets_v2.js [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+  onStateChange(fn) {
+    this.on(EVENT.STATE_CHANGE, (newState) => fn(newState));
+  }
+
+  handleMessage(data) {
+    const message = JSON.parse(data);
+    const msgType = message[0]?.T;
+
+    if (msgType == null) {
+      this.emit(ERROR.UNEXPECTED_MESSAGE);
```
Comment (emileindik) [inline lib/resources/datav2/websockets_v2.js:255]: This should be `CONN_ERROR.UNEXPECTED_MESSAGE`

**Pull request merged**